app/models/unique_deck_type.rb

```diff
@@ -30,6 +30,50 @@ def self.get_type_popularity(time_ago)
     arche.update(arche){|type, wins| wins.to_f/tot * 100 }
   end
 
+  def self.get_top_decks
+    decks = {}
+    arch_pop = Rails.cache.read('archetype_pop').sort_by{|name, val| val}.reverse
+
+    ar1 = []
+    ar2 = []
+    ar3 = []
+
+    @ar1 = UniqueDeckType.where(name: arch_pop[0][0])[0].unique_decks.where("num_matches >= ?", "30").sort_by{|ud| ud.winrate ? ud.winrate : 0 }.reverse
+    @ar1.each do |ud|
+      x = ud.decks.where(is_public: true).where("user_num_matches >= ?", "30").sort_by{|ud| ud.user_winrate ? ud.user_winrate : 0 }
```

Comment:
Use `UniqueDeckType.where(name: arch_pop[0][0])[0].unique_decks.where("num_matches >= ?", "30").sort_by{|ud| ud.winrate ? ud.winrate : 0 }.all.reverse`
so we retrieve an array.

Then use array methods `select` and such to filter through these.